jupyterhub_idle_culler/__init__.py

```diff
@@ -438,200 +442,290 @@ async def handle_user(user):
         futures.append((user["name"], handle_user(user)))
 
     if state_filter:
-        app_log.debug(f"Got {n_users} users with ready servers")
+        logger.debug(f"Got {n_users} users with ready servers")
     else:
-        app_log.debug(f"Got {n_users} users")
+        logger.debug(f"Got {n_users} users")
 
     for name, f in futures:
         try:
             result = await f
         except Exception:
-            app_log.exception(f"Error processing {name}")
+            logger.exception(f"Error processing {name}")
         else:
             if result:
-                app_log.debug("Finished culling %s", name)
+                logger.debug("Finished culling %s", name)
 
 
-def main():
-    define(
-        "url",
-        default=os.environ.get("JUPYTERHUB_API_URL"),
+class IdleCuller(Application):
+
+    api_page_size = Int(
+        0,
         help=dedent(
             """
-            The JupyterHub API URL.
+            Number of users to request per page,
+            when using JupyterHub 2.0's paginated user list API.
+            Default: user the server-side default configured page size.
             """
         ).strip(),
+    ).tag(
+        config=True,
     )
-    define(
-        "timeout",
-        type=int,
-        default=600,
+
+    concurrency = Int(
+        10,
         help=dedent(
             """
-            The idle timeout (in seconds).
+            Limit the number of concurrent requests made to the Hub.
+
+            Deleting a lot of users at the same time can slow down the Hub,
+            so limit the number of API requests we have outstanding at any given time.
             """
         ).strip(),
+    ).tag(
+        config=True,
     )
-    define(
-        "cull_every",
-        type=int,
-        default=0,
+
+    config_file = Unicode(
```

Comment:
Does this need to be added to the aliases?

Reply:
Sure, why not.  Added.